swim/test_utils.go

```diff
@@ -137,21 +155,32 @@ func destroyNodes(tnodes ...*testNode) {
 	}
 }
 
-func genAddresses(host, fromPort, toPort int) []string {
-	var addresses []string
-
+// fakeAddresses returns a slice of fake IP address/port combinations that can
+// be used during testing. Note that these addresses cannot be used in tests
+// which require real communication or binding to a socket.
+//
+// The IP addresses returned are from the TEST-NET-1 block which are defined in
+// RFC 5737 as to be used for documentation and recommended to be unrouteable.
+// In practice, these addresses may also just time out.
+//
+// See:
+// http://tools.ietf.org/html/rfc5737
+// http://stackoverflow.com/questions/10456044/what-is-a-good-invalid-ip-address-to-use-for-unit-tests
+//
+func fakeAddresses(host, fromPort, toPort int) []string {
+	var hostports []string
 	for i := fromPort; i <= toPort; i++ {
-		addresses = append(addresses, fmt.Sprintf("127.0.0.%v:%v", host, 3000+i))
+		hostports = append(hostports, fmt.Sprintf("192.0.2.%v:%v", host, i))
 	}
-
-	return addresses
+	return hostports
 }
 
-func genAddressesDiffHosts(fromPort, toPort int) []string {
-	var hosts []string
-	for i := fromPort; i <= toPort; i++ {
-		hosts = append(hosts, fmt.Sprintf("127.0.0.%v:3001", i))
+// fakeAddressesDiffHosts returns a slice of fake IP address/port combinations
+// similar to fakeAddresses but over n unique hosts (IP addresses).
+func fakeAddressesDiffHosts(fromHost, toHost int) []string {
```

Comment:
We can slam fakeAdresses and fakeAddressesDiffHosts together into one function
fakeHostPorts(fromHost, toHost, fromPort, toPort int)
Or
fakeHostPort(toHost, toPort int) //... since we always start from the first port/host anyway